app/views/unattended/provisioning_templates/snippet/preseed_netplan_setup.erb

```diff
@@ -34,7 +33,7 @@ oses:
                   :interface => bond,
                   :subnet => bond.subnet,
                   :subnet6 => bond.subnet6,
-                  :dhcp => bond.subnet.nil? ? false : bond.subnet.dhcp_boot_mode?,
+                  :dhcp => bond.subnet.nil? ? true : bond.subnet.dhcp_boot_mode?,
                   :dhcp6 => bond.subnet6.nil? ? false : bond.subnet.dhcp_boot_mode? }) -%>
```

Comment:
You are right. I think it make sense to use `bond.subnet6` instead of `bond.subnet`. Can you please change these lines of code, too?

Reply:
With @ekohl we decided there wasn't a consistent way to carry this behaviour forward. I should have closed this PR. Suggest the bond ubnet fix goes into it's own PR.